### ISSUE TITLE
Input errors cause segfault with empty err output

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#8253]
Issue overview
--------------
As noted by @Myoldmopar in https://github.com/NREL/EnergyPlus/pull/7904#issuecomment-678405079 , using an invalid object type in the idf will cause a crash with an empty err output file.

There is a message to the console:
```
EnergyPlus, Version 9.4.0-ff4b91794a, YMD=2020.09.03 13:38
**FATAL:Errors occurred on processing input file. Preceding condition(s) cause termination.
```
But then it crashes [here](https://github.com/NREL/EnergyPlus/blob/ff4b91794ae3dc5b6c17b9015d93d84122596e26/src/EnergyPlus/IOFiles.cc#L242) with an access violation.

But this all sounds familiar - yes, *any* input error (like an incorrect key choice) has the same result. This is deja vu of https://github.com/NREL/EnergyPlus/issues/8180 which produced an empty err file but didn't crash. We need a way to catch this in CI - perhaps we need a testfile that purposely has an input error?

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file: [5ZoneAirCooled-BadObject.idf.txt](https://github.com/NREL/EnergyPlus/files/5170548/5ZoneAirCooled-BadObject.idf.txt) (Bad object is "Timesteps, 4;")
[5ZoneAirCooled-BadKeyChoice.idf.txt](https://github.com/NREL/EnergyPlus/files/5170644/5ZoneAirCooled-BadKeyChoice.idf.txt)

 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by nealkruis** [comment]
@mjwitte @Myoldmopar are you looking for a champion on this one? Happy to help with it if so.

**Comment by mjwitte** [comment]
@nealkruis I'm not planning to work on this at the moment. I haven't tracked down when it broke. Sometime after #8188 was merged.

**Comment by Myoldmopar** [comment]
I don't mind if you tackle it, but of course any change in the input processor should get blessed by @mbadams5.  Any thoughts/tips @mbadams5?

**Comment by mbadams5** [comment]
I don't have any thoughts at the moment. I was going to poke around at it today since I am curious how this is happening. If I do find any leads, I'll let @nealkruis know.

**Comment by mjwitte** [comment]
Looks like @mitchute has a fix in #8273.
